config/routes.rb

```diff
@@ -3,4 +3,12 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+  get "/users/:author_id/posts", to: "posts#index"
+
+  get "/users/:author_id/posts/:id", to: "posts#show"
+
+  get "/users", to: "users#index"
+
+  get "/users/:id", to: "users#show"
```

Comment:
- [ ] [OPTIONAL] I think it would be better to use [resource routing](https://guides.rubyonrails.org/v3.2.2/routing.html#resource-routing-the-rails-default) for defining routes. It not only simplifies your code but also aligns with Rails conventions, making your application more maintainable and developer-friendly.


```ruby
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```